envs/reaching_env/reaching_task.py

```diff
@@ -179,9 +184,33 @@ def step(self, action):
         info["collision"] = collision
         return state, reward, done, info
 
+    def _get_reward_with_subgoal(self, state):
+        #### TODO: Q5 design a reward based on the state, goal and subgoal positions
+        dist_to_goal = np.linalg.norm(state - self._goal_pos[:2])
+        if (not self.subgoal_reached):
+          dist_to_subgoal1 = np.linalg.norm(state - self._subgoal2_pos[0][:2])
+          dist_to_subgoal2 = np.linalg.norm(state - self._subgoal2_pos[1][:2])
+          success_subgoal1 = dist_to_subgoal1 < 1e-3
+          success_subgoal2 = dist_to_subgoal2 < 1e-3
+
+          if (success_subgoal1 or success_subgoal2):
+            self.subgoal_reached = True
+
+        if (not self.subgoal_reached):
+          reward = (-2.0 * min(dist_to_subgoal1, dist_to_subgoal2)) - dist_to_goal
+        else:
+          reward = -1.0 * dist_to_goal
+
+        return reward
+
     def _get_reward(self, state, action, collision):
-        reward = None
         success = self.get_success(self, state)
+        if self._reward_type == "sparse":
+            reward = success
+        elif self._reward_type == "dense":
```

Comment:
can we change these to "sparse_handcrafted" and "dense_handcrafted" just to make it more clear what's really going on?